interfaces/cython/cantera/cti2yaml.py

```diff
@@ -0,0 +1,1648 @@
+#!/usr/bin/env python
+
+# This file is part of Cantera. See License.txt in the top-level directory or
+# at https://www.cantera.org/license.txt for license and copyright information.
+
+"""
+cti2yaml.py: Convert legacy CTI input files to YAML
+
+Usage:
+    python cti2yaml.py mech.cti [out.yaml]
+
+This will produce the output file 'mech.yaml' if an output file name is not
+specified.
+"""
+
+from __future__ import print_function
+
+import sys
+import os
+import re
+from collections import OrderedDict
+import numpy as np
+from email.utils import formatdate
+
+try:
+    import ruamel_yaml as yaml
+except ImportError:
+    from ruamel import yaml
+
+# Python 2/3 compatibility
+try:
+  basestring
+except NameError:
+  basestring = str
+
+
+def _printerr(*args):
+    # All debug and error output should go to stderr
+    print(*args, file=sys.stderr)
+
+
+class InputError(Exception):
+    """
+    Exception raised if an error is encountered while parsing the input file.
+    """
+    def __init__(self, msg, *args):
+        if args:
+            msg = msg.format(*args)
+        super(InputError, self).__init__(msg)
+
+BlockMap = yaml.comments.CommentedMap
+
+def FlowMap(*args, **kwargs):
+    m = yaml.comments.CommentedMap(*args, **kwargs)
+    m.fa.set_flow_style()
+    return m
+
+def FlowList(*args, **kwargs):
+    lst = yaml.comments.CommentedSeq(*args, **kwargs)
+    lst.fa.set_flow_style()
+    return lst
+
+# Improved float formatting requires Numpy >= 1.14
+if hasattr(np, 'format_float_positional'):
+    def float2string(data):
+        if data == 0:
+            return u'0.0'
```

Comment:
If I understand strings correctly in Python 3, you do not need to include the `u` because all strings are Unicode strings in Python 3. This would only be necessary if we needed to maintain Python 2 compatibility.